md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessLogDAO.java

```diff
@@ -186,19 +188,18 @@ public ProcessLogInfo getLastValue(ProcessLogInfo processLogInfo) {
         List<ProcessLogInfo> processLogInfoList = new ArrayList<ProcessLogInfo>();
 
         try {
-            Criteria listProcessLogCriteria = session.createCriteria(ProcessLog.class).setProjection(Projections.distinct(Projections.property("process.processId")));
+            Criteria listProcessLogCriteria = session.createCriteria(ProcessLog.class).setProjection(Projections.distinct(Projections.property(PARENTPROCESSID)));
             int counter = listProcessLogCriteria.list().size();
             LOGGER.info("number of distinct processid is " + counter);
             List<Integer> processList = new ArrayList<Integer>();
-            Criteria processLogListCriteria = session.createCriteria(ProcessLog.class).setProjection(Projections.distinct(Projections.property("process.processId")));
+            Criteria processLogListCriteria = session.createCriteria(ProcessLog.class).setProjection(Projections.distinct(Projections.property(PARENTPROCESSID)));
             processList = processLogListCriteria.list();
             LOGGER.info("size of processlog " + processList.size() +" and list contains:"+ processList);
-//
-                List <Process> parentProcessList=new ArrayList<>();
+
             if(processLogInfo.getProcessId()==null){
-                if (processList.size() != 0) {
-                    Criteria distintPProcessId = session.createCriteria(Process.class).add(Restrictions.in("processId",processList)).setProjection(Projections.distinct(Projections.property("process.processId")));
-                    distintPProcessId.addOrder(Order.desc("process.processId")).setFirstResult(processLogInfo.getPage()).setMaxResults(processLogInfo.getPageSize());
+                if (!processList.isEmpty()) {
+                    Criteria distintPProcessId = session.createCriteria(Process.class).add(Restrictions.in("processId",processList)).setProjection(Projections.distinct(Projections.property(PARENTPROCESSID)));
+                    distintPProcessId.addOrder(Order.desc(PARENTPROCESSID)).setFirstResult(processLogInfo.getPage()).setMaxResults(processLogInfo.getPageSize());
 
                     List<Integer>distinctPPidList=distintPProcessId.list();
                     for(Integer pPid:distinctPPidList){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)